plugin/pulpcore/plugin/download/monitor.py

```diff
@@ -0,0 +1,80 @@
+from pulpcore.download import DigestValidation, Event
+
+
+class DownloadMonitor:
+    """
+    Monitor a download and collect information:
+     - Total number of bytes downloaded.
+     - Standard set of file digests.
+
+    Attributes:
+        algorithms (dict): Dictionary of validations keyed by algorithm.
+            Used only to leverage digest calculation.
+        size (int): The total file size in bytes.
```

Comment:
It looks like this is the total number of bytes that have been received **thus far**, right?